src/network/gamehost.cc

```diff
@@ -806,95 +820,146 @@ void GameHost::send_player_command(Widelands::PlayerCommand* pc) {
  */
 void GameHost::send(ChatMessage msg) {
 	if (msg.msg.empty()) {
+		// No message: Nothing to do
 		return;
 	}
 
+	// The set containing all receivers of the message.
+	// Being a set ensures that each receiver only gets the message once
+	std::set<int32_t> recipients;
+	// Whether this is a public (0), personal (1), or team (2) message (see protocol definition)
```

Comment:
An enum class is probably too strict for it since I have to write it as integers to the network packet.
I could introduce three `const uint8_t`, if that is acceptable?

Reply:
Either constexpr or enum would work for me - the most important thing is that they have names.